x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/latest/LatestDoc.java

```diff
@@ -0,0 +1,276 @@
+import org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper;
+import org.elasticsearch.xpack.core.transform.TransformMessages;
+import org.elasticsearch.xpack.core.transform.transforms.SourceConfig;
+import org.elasticsearch.xpack.core.transform.transforms.TransformIndexerStats;
+import org.elasticsearch.xpack.core.transform.transforms.TransformProgress;
+import org.elasticsearch.xpack.core.transform.transforms.latest.LatestDocConfig;
+import org.elasticsearch.xpack.transform.transforms.Function;
+import org.elasticsearch.xpack.transform.transforms.pivot.SchemaUtil;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.function.Predicate.not;
+import static java.util.stream.Collectors.toMap;
+import static org.elasticsearch.common.xcontent.XContentFactory.jsonBuilder;
+
+public class LatestDoc implements Function {
+
+    public static final int TEST_QUERY_PAGE_SIZE = 50;
+
+    private static final String COMPOSITE_AGGREGATION_NAME = "_transform";
+    private static final String LATEST_AGGREGATION_NAME = "_latest";
+    private static final Logger logger = LogManager.getLogger(LatestDoc.class);
+
+    private final LatestDocConfig config;
+
+    // objects for re-using
+    private final CompositeAggregationBuilder cachedCompositeAggregation;
+
+    public LatestDoc(LatestDocConfig config) {
+        if (config.getSort().size() != 1) {
+            throw new ElasticsearchException("sort must specify exactly one sorting field");
+        }
+        this.config = config;
+        this.cachedCompositeAggregation = createCompositeAggregation(config);
+    }
+
+    private static CompositeAggregationBuilder createCompositeAggregation(LatestDocConfig config) {
+        assert config.getSort().size() >= 1 && config.getSort().get(0) instanceof FieldSortBuilder;
+
+        CompositeAggregationBuilder compositeAggregation;
+        try (XContentBuilder builder = jsonBuilder()) {
+            config.toCompositeAggXContent(builder);
+            XContentParser parser = builder.generator()
+                .contentType()
+                .xContent()
+                .createParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE, BytesReference.bytes(builder).streamInput());
+            compositeAggregation = CompositeAggregationBuilder.PARSER.parse(parser, COMPOSITE_AGGREGATION_NAME);
+        } catch (IOException e) {
+            throw new RuntimeException(
+                TransformMessages.getMessage(TransformMessages.TRANSFORM_FAILED_TO_CREATE_COMPOSITE_AGGREGATION, "latest_doc"), e);
+        }
+        TopHitsAggregationBuilder topHitsAgg = AggregationBuilders.topHits(LATEST_AGGREGATION_NAME).size(1).sort(config.getSort().get(0));
+        compositeAggregation.subAggregation(topHitsAgg);
+        return compositeAggregation;
+    }
+
+    @Override
+    public int getInitialPageSize() {
+        return 5000;
```

Comment:
do we have an idea about this default? I agree that we should have a higher default than pivot. 

should make this a constant which can be overridden with `config.getMaxPageSearchSize()`